### PR TITLE
LP-2377 Add experience view and apis

### PR DESCRIPTION
**Pull request by zeshanarifios**
[LP-2377](https://philanthropyu.atlassian.net/browse/LP-2377)

### CODE REVIEW

**Review by annas-usman — CHANGES_REQUESTED** [review]
Left some comments

**Review by annas-usman — CHANGES_REQUESTED** [review]

File: openedx/adg/lms/applications/migrations/0006_auto_20210116_1528.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        ),
```
Comment (ahmed-faraz46) [inline openedx/adg/lms/applications/migrations/0006_auto_20210116_1528.py:37]: Let's please remove these operation, if possible take latest pull from master, as we have fixed this issue in [LP-2498 ](https://philanthropyu.atlassian.net/browse/LP-2498)

File: openedx/adg/lms/applications/api_views.py [hunk 2]
```diff
@@ -109,10 +115,24 @@ class WorkExperienceViewSet(ApplicationRequirementsViewSetBaseClass):
...
+
+    @action(detail=False, methods=['patch'],  url_name='update_is_not_applicable', url_path='update_is_not_applicable')
+    def update_is_work_experience_not_applicable(self, request):
+        is_work_experience_not_applicable = json.loads(request.data.get('is_work_experience_not_applicable', 'false'))
+        user_application = request.user.application
+        user_application.is_work_experience_not_applicable = is_work_experience_not_applicable
+        with transaction.atomic():
+            user_application.save()
+            if is_work_experience_not_applicable:
+                WorkExperience.objects.filter(user_application=user_application).delete()
```
Comment (ahmed-faraz46) [inline openedx/adg/lms/applications/api_views.py:136]: We can change it to `user.application.applications_workexperiences.all().delete()` more django specific.

**Review by ahmed-faraz46 — CHANGES_REQUESTED** [review]

**Comment by zeshanarifios** [comment]
https://github.com/OmnipreneurshipAcademy/edx-platform/pull/143

**Pull request closed without merging**